### PR TITLE
New function CDB_CreateOverviewsWithToleranceInPixels

### LINKED ISSUE #219: OOM when generating overviews
When setting `grid_px` to 2 here: https://github.com/CartoDB/cartodb-postgresql/blob/aff7ae3e2e155de00958a8bbc7b95203ba9ffb8e/scripts-available/CDB_Overviews.sql#L557

```
irb(main):006:0* u.in_database.run("SELECT cartodb.CDB_CreateOverviews('rtorre.yellow_tripdata_2015_pickups'::regclass)")
NOTICE:  table "_vovw_19_yellow_tripdata_2015_pickups" does not exist, skipping
CONTEXT:  SQL statement "DROP TABLE IF EXISTS _vovw_19_yellow_tripdata_2015_pickups CASCADE;"
PL/pgSQL function _cdb_gridcluster_reduce_strategy(regclass,integer,integer) line 58 at EXECUTE statement
SQL statement "SELECT _cdb_gridcluster_reduce_strategy('yellow_tripdata_2015_pickups', 20, 19);"
PL/pgSQL function cdb_createoverviews(regclass,regproc,regproc) line 26 at EXECUTE statement
NOTICE:  table "_vovw_18_yellow_tripdata_2015_pickups" does not exist, skipping
CONTEXT:  SQL statement "DROP TABLE IF EXISTS _vovw_18_yellow_tripdata_2015_pickups CASCADE;"
PL/pgSQL function _cdb_gridcluster_reduce_strategy(regclass,integer,integer) line 58 at EXECUTE statement
SQL statement "SELECT _cdb_gridcluster_reduce_strategy('_vovw_19_yellow_tripdata_2015_pickups', 19, 18);"
PL/pgSQL function cdb_createoverviews(regclass,regproc,regproc) line 26 at EXECUTE statement
Sequel::DatabaseError: PG::Error: ERROR:  out of memory
DETAIL:  Failed on request of size 16072.
CONTEXT:  SQL statement "
      CREATE TABLE _vovw_18_yellow_tripdata_2015_pickups AS
         WITH clusters AS (
           SELECT
             AVG(vendorid)::integer AS vendorid,CASE count(*) WHEN 1 THEN MIN(tpep_pickup_datetime) ELSE NULL END::timestamp with time zone AS tpep_pickup_datetime,CASE count(*) WHEN 1 THEN MIN(tpep_dropoff_datetime) ELSE NULL END::timestamp with time zone AS tpep_dropoff_datetime,AVG(passenger_count)::integer AS passenger_count,AVG(trip_distance)::double precision AS trip_distance,AVG(pickup_longitude)::double precision AS pickup_longitude,AVG(pickup_latitude)::double precision AS pickup_latitude,AVG(ratecodeid)::integer AS ratecodeid,AVG(dropoff_longitude)::double precision AS dropoff_longitude,AVG(dropoff_latitude)::double precision AS dropoff_latitude,AVG(payment_type)::integer AS payment_type,AVG(fare_amount)::double precision AS fare_amount,AVG(extra)::double precision AS extra,AVG(mta_tax)::double precision AS mta_tax,AVG(tip_amount)::double precision AS tip_amount,AVG(tolls_amount)::double precision AS tolls_amount,AVG(total_amount)::double precision AS total_amount, 
             count(*) AS n,
             SUM(ST_X(f.the_geom_webmercator)) AS sx,
             SUM(ST_Y(f.the_geom_webmercator)) AS sy,
             Floor(ST_X(f.the_geom_webmercator)/0.298582144081593)::int AS gx,
             Floor(ST_Y(f.the_geom_webmercator)/0.298582144081593)::int AS gy,
             MIN(cartodb_id) AS cartodb_id
          FROM _vovw_19_yellow_tripdata_2015_pickups f
          GROUP BY gx, gy
         )
         SELECT cartodb_id,ST_Transform(ST_SetSRID(ST_MakePoint(sx/n, sy/n), 3857), 4326) AS the_geom,ST_SetSRID(ST_MakePoint(sx/n, sy/n), 3857) AS the_geom_webmercator,vendorid,tpep_pickup_datetime,tpep_dropoff_datetime,passenger_count,trip_distance,pickup_longitude,pickup_latitude,ratecodeid,dropoff_longitude,dropoff_latitude,payment_type,fare_amount,extra,mta_tax,tip_amount,tolls_amount,total_amount FROM clusters
    "
PL/pgSQL function _cdb_gridcluster_reduce_strategy(regclass,integer,integer) line 64 at EXECUTE statement
SQL statement "SELECT _cdb_gridcluster_reduce_strategy('_vovw_19_yellow_tripdata_2015_pickups', 19, 18);"
PL/pgSQL function cdb_createoverviews(regclass,regproc,regproc) line 26 at EXECUTE statement
```

### PR DESCRIPTION
**Pull request by jgoizueta**
This function allows defining the previously fixed parameter grid_px.
The lim parameter used to define the reference Z level is also
correlated to this value.

Fixes #219

### CODE REVIEW

**Comment by jgoizueta** [comment]
@ethervoid can you CR this, please?

File: scripts-available/CDB_Overviews.sql [hunk 1]
```diff
@@ -291,15 +291,21 @@ $$ LANGUAGE PLPGSQL STABLE;
 -- Parameters:
 --   reloid: oid of the input table. It must be a cartodbfy'ed table.
 -- Return value: Z level as an integer
-CREATE OR REPLACE FUNCTION _CDB_Feature_Density_Ref_Z_Strategy(reloid REGCLASS)
+CREATE OR REPLACE FUNCTION _CDB_Feature_Density_Ref_Z_Strategy(reloid REGCLASS, tolerance_px FLOAT8 DEFAULT NULL)
```
Comment (rafatower) [inline scripts-available/CDB_Overviews.sql]: isn't it easier to declare DEFAULT 500?

Comment (rafatower) [inline scripts-available/CDB_Overviews.sql]: ignore my comment

**Comment by rafatower** [comment]
:+1:

**Comment by jatorre** [comment]
I do not understand why 7.5 is the default value for this... I would assume 0.5 to be the default value (is the most correct visualization) and you can override it. But I dont see a reason/use case to set actually a value higher than 1. Happy to have it as a parameter, but default should be a reasonable number, and 7.5 does not look like it. Any reason for it?

**Pull request merged**